plugins/ingestion-kafka/src/main/java/org/opensearch/plugin/kafka/KafkaPlugin.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.kafka;
+
+import org.opensearch.index.IngestionConsumerFactory;
+import org.opensearch.plugins.IngestionConsumerPlugin;
+import org.opensearch.plugins.Plugin;
+
+import java.util.Map;
+
+/**
+ * A plugin for ingestion source of Kafka.
+ */
+public class KafkaPlugin extends Plugin implements IngestionConsumerPlugin {
```

Comment:
Would it make sense to generalize this? (Maybe `PullBasedIngestionPlugin`?) 

This use-case makes me this of the audit log use-case in the security plugin that has many different sinks: https://github.com/opensearch-project/security/tree/main/src/main/java/org/opensearch/security/auditlog/sink

Reply:
do you mean this file itself or this kafka plugin project as a whole?

Reply:
Yes, the design doc mentions supporting other types of sources in the future:

> In this proposal, we’ll use Kafka as the streaming system example, but the same ingestion capability can be applied to other similar systems like Pulsar, Kinesis, Redpanda, etc.

Do you envisage 1 plugin per source, or 1 plugin with the purpose of pull-based ingestion that supports different types of sources?

I also have a high-level question about using index-settings to setup ingestion. Curious why using index settings vs. a system index that stores configuration for this plugin and expose REST APIs to do setup?

Reply:
One other high-level question. Will these pollers run on every node of the cluster?

With the Opensearch Job Scheduler, there is a locking mechanism to ensure that any instance of a job is only run once and not run simultaneously across all nodes.

Reply:
> Do you envisage 1 plugin per source, or 1 plugin with the purpose of pull-based ingestion that supports different types of sources?

yes, there'll be one plugin per source. @msfroh will add a plugin for kinesis
> 
> I also have a high-level question about using index-settings to setup ingestion. Curious why using index settings vs. a system index that stores configuration for this plugin and expose REST APIs to do setup?

I feel index settings will be better given it'll be a core feature to opensearch